File: AutoTrade/AutoTrade.Services/Database/PaymentTransaction.cs
namespace Database
{
    public partial class PaymentTransaction
    {
        public int Id { get; set; }
        public string? PaymentId { get; set; }
        public decimal Amount { get; set; }
        public string Currency { get; set; }
        public string Status { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public int? AutomobileAdId { get; set; }
        public AutomobileAd? AutomobileAd { get; set; } 
    }
}

File: AutoTrade/Ui/desktop/lib/models/PaymentTransaction.dart
class PaymentTransaction {
  final int transactionId;
  final String paymentId;
  final double amount;
  final String currency;
  final String status;
  final DateTime createdAt;
  final DateTime updatedAt;

  PaymentTransaction({
    required this.transactionId,
    required this.paymentId,
    required this.amount,
    required this.currency,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  factory PaymentTransaction.fromJson(Map<String, dynamic> json) {
    return PaymentTransaction(
      transactionId: json['transactionId'],
      paymentId: json['paymentId'],
      amount: (json['amount'] as num).toDouble(),
      currency: json['currency'],
      status: json['status'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }
}
